### PR TITLE
Updated security report and slither output for Swap

### PR DESCRIPTION
**Pull request by dmosites**

### CODE REVIEW

File: protocols/swap/reports/SECURITY.md [hunk 1]
```diff
@@ -1,175 +1,173 @@
...
+| authorizeSigner | Swap.sol | external   | `address _authorizedSigner`, `uint256 _expiry` | no      |
+| revokeSender    | Swap.sol | external   | `address _authorizedSender`                    | no      |
+| revokeSigner    | Swap.sol | external   | `address _authorizedSender`                    | no      |
 
 ## Invariants
 
-#### 1. No ether or tokens should be held by the contract address.
+### No ether (ETH) or tokens should be held by the contract address.
 
+- No functions are payable and therefore ether (ETH) cannot be stuck by calling any functions.
```
Comment (alicevhenshaw) [inline protocols/swap/reports/SECURITY.md]: `stuck` -> `deposited`?

File: protocols/swap/reports/SECURITY.md [hunk 2]
```diff
@@ -1,175 +1,173 @@
...
+- By inspection, all branches of the function `swap` either succeed or throw. At no point can these functions return false or fail silently without throwing.
+- All transfers in the execution function are between the signer and sender or a signer and affiliate; at no point is a token transfer performed to or from the Swap contract address.
```
Comment (alicevhenshaw) [inline protocols/swap/reports/SECURITY.md]: Well, anyone could perform a token transfer to the swap contract... but that's their fault. And again it never uses its own balance. This is sort of the same point as the `seldestruct` point

Comment (alicevhenshaw) [reply protocols/swap/reports/SECURITY.md]: Oh you wrote that as the next point hahah. Okay ignore me.

File: protocols/swap/reports/SECURITY.md [hunk 3]
```diff
@@ -1,175 +1,173 @@
...
+Swap is a non-custodial exchange settlement contract. It is used in the Swap Protocol, a peer-to-peer trading protocol for Ethereum tokens. The implementation supports trading ERC-20 and ERC-721 tokens. Additional features include the ability to authorize another party to sign or send orders on one's behalf, and the ability to optionally set an affiliate that is compensated for bringing together trading parties. This contract only transfers tokens, no raw ether (ETH). The following contracts are compiled with solidity 0.5.12.
 
 ## Structure
 
-The Swap contract is comprised a contract, an interface, and a library.
+Swap includes one contract, its interface, and their dependencies.
```
Comment (alicevhenshaw) [inline protocols/swap/reports/SECURITY.md]: Again I dont like `includes`. Swap doesnt include one contract, Swap IS that one contract.

File: protocols/swap/reports/SECURITY.md [hunk 4]
```diff
@@ -1,175 +1,173 @@
...
-#### 10. Swap function either successfully execute swap or revert
+### Affiliate trades supply correct fees from signer to sender.
 
-- By manual inspection of all branches of swap and transferFrom, no functions can fail silently without causing a REVERT. In the case no REVERT branch is hit, both sides of the swap must have been successfully executed. Note that require is used in the ERC-20 case to handle tokens where a bad transfer may fail silently without reversions. This is the only external call other than safeTransferFrom, which has no return value and therefore cannot fail silently.
-- **This invariant currently holds as-is.**
+- By inspection and testing, the transfer from signer to affiliate will complete, barring issues on the underlying token contract, on line 165. In the transferToken function, when calling safeTransferFrom, a `require` is not required for safeTransferFrom as the contract signature guarantees there is no return parameter.
+- **This invariant holds as-is.**
 
-## Analysis
+### The `cancel` function will set all provided nonces to UNAVAILABLE.
```
Comment (alicevhenshaw) [inline protocols/swap/reports/SECURITY.md]: Is this an invariant? or is it just the performance of a function?

File: protocols/swap/reports/SECURITY.md [hunk 5]
```diff
@@ -1,175 +1,173 @@
...
-#### 3. All flows are identical (no calls to underlying token contract) until the internal transferToken call.
+### An order cannot be settled at any time beyond the `expiry` specified on the order.
 
-- In the ERC-20 case, reverts will occur on the underlying transferFrom implemented methods. Applications using this contract may want to do pre-checks on known tokens to do better user experience.
-- **This invariant currently holds as-is.**
+- The `expiry` of each order is checked in the first line of the `swap` function, comparing the `expiry` to `block.timestamp` and throwing if at or beyond it.
+- **This invariant holds as-is.**
 
-#### 4. Signatures cannot be forged or duplicated.
+### Signatures cannot be forged or duplicated.
```
Comment (alicevhenshaw) [inline protocols/swap/reports/SECURITY.md]: Feels like the following 2 invariants could potentially be merged? They certainly have overlap.
1) Signatures cannot be forged or duplicated.
2) Orders can only be successfully taken once.
Maybe 2 is a subset of 1?

Comment (dmosites) [reply protocols/swap/reports/SECURITY.md]: Updated the first one to be more signature-centric.

**Pull request merged**